src/python/pants/backend/docker/rules.py

```diff
@@ -0,0 +1,121 @@
+# Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+import logging
+import os
+from dataclasses import dataclass
+
+from pants.backend.docker.rules_binary import DockerBinary, DockerBinaryRequest
+from pants.backend.docker.rules_binary import rules as binary_rules
+from pants.backend.docker.rules_context import DockerBuildContext, DockerBuildContextRequest
+from pants.backend.docker.rules_context import rules as context_rules
+from pants.backend.docker.target_types import (
+    DockerContextRoot,
+    DockerImageSources,
+    DockerImageVersion,
+)
+from pants.core.goals.package import BuiltPackage, BuiltPackageArtifact, PackageFieldSet
+from pants.engine.fs import Digest, Snapshot
+from pants.engine.process import Process, ProcessResult
+from pants.engine.rules import Get, collect_rules, rule
+from pants.engine.target import TransitiveTargets, TransitiveTargetsRequest
+from pants.engine.unions import UnionRule
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True)
+class DockerFieldSet(PackageFieldSet):
+    required_fields = (DockerImageSources,)
+
+    context_root_field: DockerContextRoot
+    image_version: DockerImageVersion
```

Comment:
Eventually we're going to want to support dynamic version tags, either autoincremented or computed from a hash. But for now this is great.

Reply:
Agreed.